<a href="https://colab.research.google.com/github/DhananiParita/Loan_risk_prediction/blob/main/project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Loan Risk Prediction Dataset Cleaning Project**

This project focuses on cleaning a Loan Risk Prediction dataset using Python and pandas in Google Colab. The dataset contains information related to customers such as age, income, loan amount, credit score, work experience, and loan approval status.

The following tools and libraries were used:
- Google Colab
- Python
- pandas library
- NumPy library

In [1]:
from google.colab import files
uploaded = files.upload()

Saving loan_risk_prediction_dataset.csv to loan_risk_prediction_dataset.csv


In [14]:
import pandas as pd
import numpy as np

In [16]:
df = pd.read_csv("loan_risk_prediction_dataset.csv")

In [5]:
df.head()

,Age,Income,LoanAmount,CreditScore,YearsExperience,Gender,Education,City,EmploymentType,LoanApproved
0,56,48353.0,31258.0,675.0,20,Female,High School,Houston,Unemployed,0
1,69,57462.0,23262.0,586.0,6,Male,High School,San Francisco,Self-Employed,0
2,46,44219.0,26530.0,781.0,26,Male,PhD,Houston,Self-Employed,1
3,32,56307.0,11531.0,549.0,11,Male,NaN,New York,Unemployed,0
4,60,37034.0,27871.0,500.0,19,Female,High School,Chicago,Unemployed,0


In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Age              5000 non-null   int64  
 1   Income           4804 non-null   float64
 2   LoanAmount       5000 non-null   float64
 3   CreditScore      4806 non-null   float64
 4   YearsExperience  5000 non-null   int64  
 5   Gender           5000 non-null   object 
 6   Education        4802 non-null   object 
 7   City             5000 non-null   object 
 8   EmploymentType   5000 non-null   object 
 9   LoanApproved     5000 non-null   int64  
dtypes: float64(3), int64(3), object(4)
memory usage: 390.8+ KB


In [18]:
df.describe()

,Age,Income,LoanAmount,CreditScore,YearsExperience,LoanApproved
count,5000.000000,4804.000000,5000.000000,4806.000000,5000.000000,5000.000000
mean,43.584600,49738.123022,19870.768600,575.494590,19.599000,0.230200
std,14.919094,15101.361851,8046.542413,160.550839,11.516837,0.421003
min,18.000000,-3731.000000,-10059.000000,300.000000,0.000000,0.000000
25%,31.000000,39608.500000,14455.250000,433.000000,10.000000,0.000000
50%,43.000000,49488.000000,19842.500000,579.000000,20.000000,0.000000
75%,56.000000,59917.000000,25326.750000,712.000000,29.000000,0.000000
max,69.000000,99146.000000,48353.000000,849.000000,39.000000,1.000000


In [19]:
df.isnull().sum()

,0
Age,0
Income,196
LoanAmount,0
CreditScore,194
YearsExperience,0
Gender,0
Education,198
City,0
EmploymentType,0
LoanApproved,0


| Problem              | Found?                 |
| -------------------- | ---------------------- |
| Missing values       | Check using `isnull()` |
| Duplicate rows       | Need to check          |
| Negative income      | Yes                    |
| Negative loan amount | Yes                    |
| Wrong formatting     | Need checking          |
| Wrong data types     | Need checking          |


The dataset was checked for common data quality issues such as missing values, duplicate rows, invalid values, and incorrect data types.

The following problems were identified:
1. Negative values in Income column
2. Negative values in LoanAmount column
3. Need to verify duplicates and missing values

In [20]:
df.duplicated().sum()

np.int64(0)

In [21]:
df.drop_duplicates(inplace=True)

In [22]:
df.isnull().sum()

,0
Age,0
Income,196
LoanAmount,0
CreditScore,194
YearsExperience,0
Gender,0
Education,198
City,0
EmploymentType,0
LoanApproved,0


In [23]:
df.fillna(df.mean(numeric_only=True), inplace=True)

In [24]:
df = df[df['Income'] >= 0]
df = df[df['LoanAmount'] >= 0]

In [25]:
df.dtypes

,0
Age,int64
Income,float64
LoanAmount,float64
CreditScore,float64
YearsExperience,int64
Gender,object
Education,object
City,object
EmploymentType,object
LoanApproved,int64


In [26]:
df['Age'] = df['Age'].astype(int)

The following cleaning steps were performed on the dataset:

1. Imported dataset using pandas
2. Viewed dataset information using head(), info(), and describe()
3. Checked missing values using isnull()
4. Checked duplicate rows using duplicated()
5. Removed invalid negative values from Income and LoanAmount columns
6. Verified data types of all columns
7. Saved the cleaned dataset

In [27]:
df.describe()

,Age,Income,LoanAmount,CreditScore,YearsExperience,LoanApproved
count,4969.000000,4969.000000,4969.000000,4969.000000,4969.000000,4969.000000
mean,43.570739,49763.052347,20002.458040,575.229817,19.590662,0.230831
std,14.923863,14724.129702,7884.595787,157.404280,11.521067,0.421407
min,18.000000,129.000000,4.000000,300.000000,0.000000,0.000000
25%,31.000000,40078.000000,14546.000000,439.000000,10.000000,0.000000
50%,43.000000,49738.123022,19885.000000,575.494590,20.000000,0.000000
75%,56.000000,59426.000000,25374.000000,705.000000,29.000000,0.000000
max,69.000000,99146.000000,48353.000000,849.000000,39.000000,1.000000


Before cleaning:
- Income minimum value = -3731
- LoanAmount minimum value = -10059
- Total rows = 5000

After cleaning:
- Income minimum value = 129
- LoanAmount minimum value = 4
- Total rows = 4969

A total of 31 invalid rows were removed from the dataset.

In [28]:
df.to_csv("cleaned_loan_dataset.csv", index=False)

In [29]:
from google.colab import files
files.download("cleaned_loan_dataset.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

The dataset was successfully cleaned and improved by removing unrealistic negative values and verifying overall data consistency. The cleaned dataset is now ready for further data analysis and machine learning tasks.